File: we-care-app-frontend/src/app/common-components/models/doctor-dto.ts
export interface DoctorDTO {
    id?: number,
    firstName: string,
    lastName: string,
    briefIntroduction: string,
    joiningDate: string,
    doctorImageURL?: string,
    totalRating: number,
    averageRating: number,
    specialityName: string,
    fees: number,
    clinicStates: string[] ;
    clinicCities: string[];
}

File: we-care-app-backend/src/main/java/clinicmangement/com/WeCare/DTO/DoctorDTO.java
package clinicmangement.com.WeCare.DTO;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class DoctorDTO {


    private Integer id;
    private String firstName;
    private String lastName;

    private String briefIntroduction;
    private LocalDateTime joiningDate;


    private MultipartFile medicalCardFile;
    private String medicalCardURL;

    private MultipartFile doctorImgFile;
    private String doctorImageURL;

    private Double totalRating;

    private Double averageRating;

    private Integer userId;

    private Integer specialityId;

    private String specialityName;

    private Integer fees;

    private Long lastUpdated;

    private List<String> clinicStates;

    private List<String> clinicCities;

    private List<ScheduleDTORead> scheduleDTOReads;




}
